enrichments/trace/internal/elastic/span.go

```diff
@@ -319,6 +356,9 @@ func isElasticTransaction(span ptrace.Span) bool {
 	return false
 }
 
+// getHostPort derives the host:port value from url.* attributes. Unlike
+// apm-data, the current code does NOT fallback to net.* or http.*
+// attritubtes as most of these are now deprecated.
 func getHostPort(urlFull *url.URL, urlDomain string, urlPort int64) string {
```

Comment:
[**For reviewers**] The handling for destination host and port have a few variants in apm-data in terms of fallback but I believe at-least the current complexity of handling this in apm-data is not worth the effort and should be fixed. Below are the main common points:

1. Both, destination port and address, are derived from `url.full` (apm-data uses the now deprecated `http.url`) ([ref](https://github.com/elastic/apm-data/blob/3fdb8a57c602f9d9c49853a1d20fc532458ae21d/input/otlp/traces.go#L865-L905))
2. First level of fallback in apm-data uses `http.*` attributes to build the full url. Unfortunately, these `http.*` attributes are also deprecated (and some deprecations are present from quite some time). In the current enrichments code, the fallback uses the latest `url.*` attributes which replace the deprecated `http.*` ones.
3. APM-data has another level of fallback which uses `net.peer.{port, name, ip}`, all of these attributes are now deprecated and some are even removed from the latest semconv definitions. I have not implemented this fallback level at all in the enrichments.

Let me know if any of the above doesn't make sense, I will add the fallbacks if needed. The apm-data code can be perused [here](https://github.com/elastic/apm-data/blob/3fdb8a57c602f9d9c49853a1d20fc532458ae21d/input/otlp/traces.go#L865-L905).

Reply:
++ Seems reasonable. As mentioned in the meeting a couple of days ago, those deprecations would ideally be handled by the [schema processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/schemaprocessor). Perhaps if we find we need to handle the deprecated attributes, we could implement some minimal transformation just prior to enrichment.